src/Passes/Source/AllocationManager/AllocationManager.cpp

```diff
@@ -61,9 +63,21 @@ namespace quantum
             throw std::runtime_error("Qubit segment not found.");
         }
 
-        mappings_.erase(it);
-        if (reuse_qubits_)
+        if (!reuse_qubits_)
+        {
+            mappings_.erase(it);
+        }
+        else
         {
+            if (it->size > registersInUse())
```

Comment:
Does the allocation manager rely on consecutive ordering of the qubit indices for qubits that are currently in use?

Reply:
It does not rely on consecutive, but it operates optimally with FILO/stack-based qubit allocation. That said, if you have another scheme that you prefer, you can use the interface to write a new manager to replace this manager. We just need to add the external interface to adding that but that is relatively easy,